sound/soc/sof/ipc.c

```diff
@@ -46,6 +46,16 @@ struct snd_sof_ipc {
 	struct list_head empty_list;
 };
 
+struct sof_ipc_ctrl_data_params {
+	u32 msg_bytes;
+	u32 hdr_bytes;
+	u32 num_msg;
+	u32 elems;
+	u32 pl_size;
```

Comment:
I really don't think it's a good idea to use size-fixed types for generic variables. I haven't checked all these variables, but I think most of them should be size_t or "unsigned int" or possibly, if negative values are also allowed, "int." msg_bytes and hdr_bytes definitely look like they should be size_t